src/modules/rlm_sql/drivers/rlm_sql_postgresql/rlm_sql_postgresql.c

```diff
@@ -326,10 +326,16 @@ static CC_HINT(nonnull) sql_rcode_t sql_query(rlm_sql_handle_t *handle, UNUSED r
 	while (PQisBusy(conn->db)) {
 		FD_ZERO(&read_fd);
 		FD_SET(sockfd, &read_fd);
-		if (!select(sockfd + 1, &read_fd, NULL, NULL, config->query_timeout ? &timeout : NULL)) {
+		r = select(sockfd + 1, &read_fd, NULL, NULL, config->query_timeout ? &timeout : NULL);
+		if (r == 0) {
 			ERROR("rlm_sql_postgresql: Socket read timeout after %d seconds", config->query_timeout);
 			return RLM_SQL_RECONNECT;
 		}
+		if (r < 0) {
+			if (errno == EINTR) continue;
```

Comment:
Continuing on EINTR, otherwise failing seems to be the majority case elsewhere in the code.

Reply:
Yes, that's certainly what I usually do when using select.  I don't think a new release is imminent. Could pull in the changes and then ask for volunteers to test on the user's list :)

Reply:
I'll be around for the next few days, so if it breaks things then I'll own it :-)

Reply:
Ah, and you need to sign your commits.

Reply:
or at least the last commit.

Reply:
Oops. Thought I'd set that to auto on the repo. Force-pushed.